### PR TITLE
Fix 4 pol remove degen

### PR DESCRIPTION
**Pull request by jsdillon**
Designed to address the need for 4-pol degeneracy removal (issue #50) as highlighted by HERA Memo 30.

### CODE REVIEW

**Comment by jsdillon** [comment]
Still need to add testing for new version of remove_degen.

File: hera_cal/omni.py [hunk 1]
```diff
@@ -332,9 +332,91 @@ def aa_to_info(aa, pols=['x'], fcal=False, minV=False, **kwargs):
...
+    antpos = info.get_antpos()
+    if (len(antpols) != 2) or (not (len(pols) == 3 and minV) 
+                               and not (len(pols) == 4 and not minV)):
+        raise ValueError('Trying to use removedegen_4pol without 4-pol data.')
+
+    # Gains loop over antenna fastest, then antpol.
+    ants = {antpol: [ant for ant in g[antpol].keys()] for antpol in antpols}
+    gains = np.array([g[antpol][ant] / g0[antpol][ant] for antpol in antpols 
+                      for ant in ants[antpol]])
+    gainPols = np.array([antpol for antpol in antpols for ant in ants[antpol]])
```
Comment (SaulAryehKohn) [inline hera_cal/omni.py]: It would be useful to leave a comment about what "gainpols" are.

File: hera_cal/omni.py [hunk 2]
```diff
@@ -332,9 +332,91 @@ def aa_to_info(aa, pols=['x'], fcal=False, minV=False, **kwargs):
...
+    if minV:
+        # In this case, there's only one phase term which should sum to 0
+        Rgains = np.hstack((positions, np.ones((positions.shape[0],1))))
+        Rvis = np.hstack((-bl_vecs, np.zeros((len(bl_vecs),1))))
+    else:
+        # In this case, x and y phases get their own columns (separate degens)
+        phaseTerms = np.vstack((gainPols==antpols[0],gainPols==antpols[1])).T
+        Rgains = np.hstack((positions, phaseTerms))
+        Rvis = np.hstack((-bl_vecs, np.zeros((len(bl_vecs),2))))    
+    Mgains = np.linalg.pinv(Rgains.T.dot(Rgains)).dot(Rgains.T)
```
Comment (SaulAryehKohn) [inline hera_cal/omni.py]: Could you explain what's happening on lines 398 & 399. It's compact and important!

File: hera_cal/omni.py (hunk 1 above)
Comment (jsdillon) [inline hera_cal/omni.py]: I've added a comment in redcal to explain this.

File: hera_cal/omni.py (hunk 2 above)
Comment (jsdillon) [inline hera_cal/omni.py]: I've added a comment in redcal to explain this.

**Comment by jsdillon** [comment]
@SaulAryehKohn I've added a bunch of comments to clarify what's going on in `remove_degen` (which now lives in redcal.py and is wrapped by omni.py). Also, in response to your suggestion, I've unified remove_degen to handle all 4 polarization cases (`1pol`, `2pol`, `4pol`, and `4pol_minV`) and am using redcal's `remove_degen` for all 4 in omni.py as well as redcal.py.

**Comment by SaulAryehKohn** [comment]
@jsdillon sorry I'm only getting to this now. Thanks for the comments and the changes. I like the explicit split between `pol_reds` and `pos_reds`, too.

**Pull request merged**